app/util/appUtil.js

```diff
@@ -337,14 +337,35 @@ async function getItem(tokenId) {
 
   if (tokenId) {
     await api.isReady
-    const item = await api.query.simpleNftModule.tokensById(tokenId)
+    const item = (await api.query.simpleNftModule.tokensById(tokenId)).toJSON()
+    const timestamp = await getTimestamp(item.created_at)
 
-    response = item.toJSON()
+    response = { ...item, timestamp }
   }
 
   return response
 }
 
+async function getTimestamp(blockNumber) {
+  await api.isReady
+  const hash = await api.rpc.chain.getBlockHash(blockNumber)
+  const block = await api.rpc.chain.getBlock(hash)
+
+  const extrinsics = block.block.extrinsics.toHuman()
+  const timestampEx = extrinsics.find(({ method: { section } }) => section === 'timestamp')
+
+  if (!timestampEx) return null
```

Comment:
shall we return some default date? just a suggestion? Do we handle return `null`? e.g. returning some error or just not including the property?

Reply:
`null` seems appropriate to me since it’s telling the API user, the block for this item/token has no timestamp on it. I don’t think it’s a 500 since no timestamp on a block isn’t an error